### ISSUE TITLE
[Request] Converting coco keypoints to open pose body keypoints ?

### ISSUE THREAD

**Issue by basicvisual** [stefanopini/simple-HRNet#68]
Hi i was wondering if there is a possibility to convert the COCO keypoints to Open Pose Body 18 keypoints.  It seems the openpose key points  have 18 keypoints with a different order ?  If there is a possibility to convert them , it could be useful as many other applications take openpose keypoints as input rather than coco keypoints. 

What are your views?

**Comment by stefanopini** [comment]
Hi! I'm sorry for the late reply.

I've checked and OpenPose, in the 19-keypoint version (i.e. without additional keypoints for feet), provides the same keypoints of COCO plus an additional mid hip joint. Moreover, HRNet has been trained on COCO without the neck joint.
Thus, there are two missing joints in the output of HRNet, but they can be computed as the mean point between the right and the left hip and the right and the left shoulder.

You can check

- the OpenPose joints [here](https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/41d8c087459fae844e477dda50a6f732e70f2cb8/src/openpose/pose/poseParameters.cpp), among many others, and [here](https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/20d8eca4b43fe28cefc02d341476b04c6a6d6ff2/doc/output.md)

- the COCO joints [here](https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/41d8c087459fae844e477dda50a6f732e70f2cb8/src/openpose/pose/poseParameters.cpp#L7), ordered as in OpenPose, and [here](https://github.com/stefanopini/simple-HRNet/blob/master/misc/visualization.py), ordered as (simple-)HRNet (neck excluded).

Bottom line, it seems doable and should be easy.
I've added this into the To Do list 🙂

**Comment by Mystickal** [comment]
@stefanopini
Hi, appreciate the work you've done here! I've got a question about having it plot 18 keypoints, wondering if you've had any luck in adding it or have an insight as to a good place to start and modify the code?

Thanks in advance!

**Comment by stefanopini** [comment]
Hi @Mystickal , sorry for the late reply.

Unfortunately, I haven't found the time to work on this yet.

In SimpleHRNet.py, you will have access to all the keypoints after [these two for loops](https://github.com/stefanopini/simple-HRNet/blob/ddc746f58b2de6f625a4cc2c4c501e8103f471f3/SimpleHRNet.py#L243-L255), so you can create an extended version of `pts` with the additional keypoints and compute them.
Then, you can add a new joint type in the `joints_dict` you find in [visualization.py](https://github.com/stefanopini/simple-HRNet/blob/master/misc/visualization.py) and use it when calling the drawing functions (e.g. [here](https://github.com/stefanopini/simple-HRNet/blob/ddc746f58b2de6f625a4cc2c4c501e8103f471f3/scripts/live-demo.py#L117)).

Let me know if this is enough to start modifying the code!
I'll try to work on this change within the next few weekends.
